src/codegen/function_body_chunk.rs

```diff
@@ -474,7 +474,13 @@ impl Builder {
                         match rust_type(env, trampoline.ret.typ).unwrap().as_str() {
                             "GString" |
                             "File" => "res.to_glib_full()".to_owned(),
-                            _ => "res.to_glib()".to_owned(),
+                            _ => {
+                                if env.type_(trampoline.ret.typ).is_fundamental_type(env) {
```

Comment:
Are enums, flags and other scalar types also considered "fundamental"?

Reply:
No.

Reply:
Then this is wrong and you need to match more on the actual type (what is returned by `env.type_()`).

There should be a function already for figuring this out.

Reply:
`impl TrampolineToGlib for library::Parameter` maybe, or code related to that

Reply:
Ok!